packages/components/src/toggle-group-control/test/index.js

```diff
@@ -38,4 +53,36 @@ describe( 'ToggleGroupControl', () => {
 		fireEvent.click( firstRadio );
 		expect( mockOnChange ).toHaveBeenCalledWith( 'rigas' );
 	} );
+	it( 'should render tooltip where `showTooltip` === `true`', async () => {
+		render(
+			<ToggleGroupControl label="Test Toggle Group Control">
+				{ optionsWithTooltip }
+			</ToggleGroupControl>
+		);
+		const firstRadio = screen.getByLabelText(
+			'Click for Delicious Gnocchi'
+		);
+		fireEvent.mouseOver( firstRadio );
+		await waitFor( () =>
+			expect(
+				screen.getByText( 'Click for Delicious Gnocchi' )
+			).toBeInTheDocument()
+		);
+	} );
+	it( 'should not render tooltip', async () => {
+		render(
+			<ToggleGroupControl label="Test Toggle Group Control">
+				{ optionsWithTooltip }
+			</ToggleGroupControl>
+		);
+		const secondRadio = screen.getByLabelText(
+			'Click for Sumptuous Caponata'
+		);
+		fireEvent.mouseOver( secondRadio );
+		await waitFor( () =>
+			expect(
+				screen.queryByText( 'Click for Sumptuous Caponata' )
+			).not.toBeInTheDocument()
+		);
+	} );
```

Comment:
Since the tooltip does not show immediately, "waiting for the tooltip text not to be in the document" will aways be a successful test (even when the tooltip should show).

Probably the easiest way could be to use fake timers ([RTL docs](https://testing-library.com/docs/using-fake-timers/), [Jest docs](https://jestjs.io/docs/next/timer-mocks))

Looks like you could do something like:

```js
it('test name', () => {
  jest.useFakeTimers();

  // do stuff, up until firing mouseOver event

  jest.runAllTimers();

  // Check for the tooltip text not to be in the document

  jest.runOnlyPendingTimers();
  jest.useRealTimers()
}
```

Although I've never used these APIs before, so I'm not sure if they will work as expected!

Reply:
> Since the tooltip does not show immediately, "waiting for the tooltip text not to be in the document" will aways be a successful test (even when the tooltip is enabled).

Thanks @ciampo !

I should have looked at the [Tooltip tests](https://github.com/WordPress/gutenberg/blob/21c9f042c3c0f44ee6147809d74291b67724a395/packages/components/src/tooltip/test/index.js) for guidance too, where it uses the fake timers. It's much neater now!